src/torchmetrics/functional/clustering/utils.py

```diff
@@ -0,0 +1,77 @@
+# Copyright The Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Optional
+
+import torch
+from torch import Tensor
+
+from torchmetrics.utilities.checks import _check_same_shape
+
+
+def calculate_contingency_matrix(
```

Comment:
can we also add a `tests/unittests/clustering/test_utils.py` file and test this function directly?
we can try this testcase: https://github.com/scikit-learn/scikit-learn/blob/1a9c0067b9320736f7e0477c86168e9fdc195253/sklearn/metrics/cluster/tests/test_supervised.py#L263-L270
but probably also add some random test cases